### ISSUE TITLE
test failure k8s.io/kubernetes/pkg/cloudprovider/providers/gce: "google: could not find default credentials"

### ISSUE THREAD

**Issue by pohly** [kubernetes/kubernetes#59922]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

```
$ make test GOFLAGS='-v -run TestEnsureExternalLoadBalancer' WHAT=k8s.io/kubernetes/pkg/cloudprovider/providers/gce
...
=== RUN   TestEnsureExternalLoadBalancer
--- FAIL: TestEnsureExternalLoadBalancer (0.01s)
        Error Trace:    gce_loadbalancer_external_test.go:252
	Error:      	Received unexpected error:
	            	google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
FAIL
exit status 1
FAIL	k8s.io/kubernetes/pkg/cloudprovider/providers/gce	0.031s
Makefile:182: recipe for target 'test' failed
```

Several other TestEnsureExternalLoadBalancer* tests fail the same way.

**What you expected to happen**:

Unit testing should pass.

**How to reproduce it (as minimally and precisely as possible)**:

See above.

**Anything else we need to know?**:

I started bisecting, but already the very first commit introducing this test (6fedaa88ec764d8442ca2ec7bf9b0d60eed447e2) fails for me, so it looks like this test has had this issue from the beginning.

**Environment**:
- Kubernetes version (use `kubectl version`): current master, 
- OS (e.g. from /etc/os-release): Debian GNU/Linux 9 (stretch)
- Kernel (e.g. `uname -a`): 4.9.0-5-amd64

/cc @agau4779

**Comment by pohly** [comment]
strace shows that the test looks for `.config/gcloud/application_default_credentials.json` in the user's home directory.

That comes from FindDefaultCredentials in https://github.com/kubernetes/kubernetes/blob/master/vendor/golang.org/x/oauth2/google/default.go#L64. It gets called by:
```
(gdb) where
#0  k8s.io/kubernetes/vendor/golang.org/x/oauth2/google.FindDefaultCredentials (ctx=..., scope= []string = {...}, ~r2=0x0, ~r3=...)
    at /fast/work/gopath/src/k8s.io/kubernetes/vendor/golang.org/x/oauth2/google/default.go:80
#1  0x000000000071a2b7 in k8s.io/kubernetes/vendor/golang.org/x/oauth2/google.DefaultTokenSource (ctx=..., scope= []string = {...}, ~r2=..., ~r3=...)
    at /fast/work/gopath/src/k8s.io/kubernetes/vendor/golang.org/x/oauth2/google/default.go:43
#2  0x0000000001547993 in k8s.io/kubernetes/pkg/cloudprovider/providers/gce.newOauthClient (tokenSource=..., ~r1=0x0, ~r2=...)
    at /fast/work/gopath/src/k8s.io/kubernetes/pkg/cloudprovider/providers/gce/gce.go:834
#3  0x000000000159cc48 in k8s.io/kubernetes/pkg/cloudprovider/providers/gce.fakeGCECloud (~r0=0x4f61d0 <testing.tRunner>, ~r1={void (void)} 0xc42006de60, ~r2=...)
    at /fast/work/gopath/src/k8s.io/kubernetes/pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go:267
#4  0x000000000159db57 in k8s.io/kubernetes/pkg/cloudprovider/providers/gce.TestEnsureExternalLoadBalancer (t=0xc4201641e0)
    at /fast/work/gopath/src/k8s.io/kubernetes/pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go:385
#5  0x00000000004f62a0 in testing.tRunner (t=0xc4201641e0, fn={void (struct testing.T *)} 0xc42006dfd8) at /usr/lib/go-1.9/src/testing/testing.go:746
#6  0x000000000045e351 in runtime.goexit () at /usr/lib/go-1.9/src/runtime/asm_amd64.s:2337
```

Looks like an empty file at that place is not enough, I still get the same error after the test opens the file for reading.

I also tried creating a fake GOOGLE_APPLICATION_CREDENTIALS file in fakeGCECloud, using the code from https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/plugin/pkg/client/auth/gcp/gcp_test.go#L177. That gets me past the initial error, but the test then hangs with:
```
E0219 16:08:53.375402    7040 gce.go:848] error fetching initial token: private key should be a PEM or plain PKSC1 or PKCS8; parse error: asn1: syntax error: sequence truncated
E0219 16:08:58.375699    7040 gce.go:848] error fetching initial token: private key should be a PEM or plain PKSC1 or PKCS8; parse error: asn1: syntax error: sequence truncated
E0219 16:09:03.375668    7040 gce.go:848] error fetching initial token: private key should be a PEM or plain PKSC1 or PKCS8; parse error: asn1: syntax error: sequence truncated
...
```

Any ideas what that means and what  really needs to be in the GOOGLE_APPLICATION_CREDENTIALS file?

**Comment by pohly** [comment]
This was fixed by a9e0a2b09ac5c8481a2de51d12b43c622ed21c05.
